pkg/remote/codec/protobuf/grpc.go

```diff
@@ -41,11 +46,28 @@ type protobufV2MsgCodec interface {
 	XXX_Marshal(b []byte, deterministic bool) ([]byte, error)
 }
 
-type grpcCodec struct{}
+type grpcCodec struct {
+	ThriftCodec remote.PayloadCodec
+}
+
+type CodecOption func(c *grpcCodec)
+
+func WithThriftCodec(t remote.PayloadCodec) CodecOption {
+	return func(c *grpcCodec) {
+		c.ThriftCodec = t
+	}
+}
 
 // NewGRPCCodec create grpc and protobuf codec
-func NewGRPCCodec() remote.Codec {
-	return new(grpcCodec)
+func NewGRPCCodec(opts ...CodecOption) remote.Codec {
```

Comment:
could we move grpcCodec to grpc package? protobuf.WithThriftCodec is weird

Reply:
It requires changing of exported symbol `NewGRPCCodec`, and also `protobuf.protobufMsgCodec` have to be exported, is it ok?

Reply:
Done